python/src/aac/plugins/rest_api/aac_rest_app.py

```diff
@@ -1,22 +1,113 @@
 """Module for configuring and maintaining the restful application and its routes."""
-from dataclasses import dataclass
-from fastapi import FastAPI, HTTPException
+from fastapi import FastAPI, HTTPException, BackgroundTasks
 from http import HTTPStatus
-import yaml
-
+import os
+import logging
+
+from aac.io.files.aac_file import AaCFile
+from aac.io.files.find import find_aac_files, is_aac_file
+from aac.io.paths import sanitize_filesystem_path
+from aac.io.parser import parse
+from aac.io.writer import write_definitions_to_file
 from aac.lang.active_context_lifecycle_manager import get_active_context
-from aac.lang.definitions.definition import Definition
-from aac.plugins.rest_api.definition_model import DefinitionModel, to_definition_class, to_definition_model
+from aac.plugins.rest_api.models.definition_model import DefinitionModel, to_definition_class, to_definition_model
+from aac.plugins.rest_api.models.file_model import FileModel, FilePathModel, FilePathRenameModel, to_file_model
 
 app = FastAPI()
 
+AVAILABLE_AAC_FILES: list[AaCFile] = []
+WORKSPACE_DIR: str = os.getcwd()
+
+
+# File CRUD Operations
 
-@app.get("/files", status_code=HTTPStatus.OK)
-def get_files():
+@app.get("/files/context", status_code=HTTPStatus.OK, response_model=list[FileModel])
+def get_files_from_context():
     """Return a list of all files contributing definitions to the active context."""
-    file_names = [definition.source_uri for definition in get_active_context().definitions]
-    return {"files": set(file_names)}
+    return [to_file_model(file) for file in get_active_context().get_files_in_context()]
+
+
+@app.get("/files/available", status_code=HTTPStatus.OK, response_model=list[FileModel])
+def get_available_files(background_tasks: BackgroundTasks):
+    """Return a list of all files available in the workspace for import into the active context. The list of files returned does not include files already in the context."""
```

Comment:
Something seems off, here, and I'm unsure if it's the docstring or the execution. As written, I'd expect this to return several files that can be imported into the context but the first time I make this request, I get an empty list as output (after starting the rest server from the `aac/python` directory).

This seems to be a result of the fact that the `refresh_available_files_in_workspace` function doesn't complete before returning because calling it again, I get the expected output. I'm not too familiar with `BackgroundTasks` but if it's an asynchronous function, do we need to `await` it so it completes before running?

Alternatively, feel free to point out what is wrong with my call.

Reply:
I added a line `asyncio.run(refresh_available_files_in_workspace())` to the server start up function, so the initial value won't be an empty list, however, it does take noticeably longer to start up. I think that issue is because the `is_aac_file` check is parsing the file. We could speed up that method by providing a quick parse that doesn't calculate and set lexemes.